lib/install/management/dispatcher.go

```diff
@@ -245,3 +254,47 @@ func (d *Dispatcher) CollectDiagnosticLogs() {
 		}
 	}
 }
+
+func (d *Dispatcher) DeleteVCH(conf *metadata.VirtualContainerHostConfigSpec) error {
+	log.Infof("Removing VMs")
+
+	if err := d.DeleteVCHInstances(conf); err != nil {
+		return err
+	}
+	if err := d.destroyResourcePoolIfEmpty(conf); err != nil {
+		return err
+	}
+	return d.removeNetwork(conf)
+}
+
+func (d *Dispatcher) DeleteVCHInstances(conf *metadata.VirtualContainerHostConfigSpec) error {
+	var errs []string
+
+	var err error
+	var vmm *vm.VirtualMachine
+	var children []*vm.VirtualMachine
+
+	if vmm, err = d.findAppliance(conf); err != nil {
```

Comment:
Added one method to search appliance VM through ID, instead of name. But will wait your PR, and then switch to that one.
I don't want to have more conflict with you.